src/app/features/event/event-form/hashtags.component.ts

```diff
@@ -0,0 +1,152 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { COMMA, ENTER } from '@angular/cdk/keycodes';
+import { ElementRef, ViewChild } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import {
+  MatAutocompleteSelectedEvent,
+  MatAutocompleteTrigger,
+} from '@angular/material/autocomplete';
+import { MatChipEditedEvent, MatChipInputEvent } from '@angular/material/chips';
+import { Observable } from 'rxjs';
+import { map, startWith } from 'rxjs/operators';
+import { inject } from '@angular/core';
+import { SnackBarService } from '@shared/services/snack-bar.service';
+
+@Component({
+  selector: 'app-hashtags',
+  templateUrl: 'hashtag.component.html',
+  styles: [
+    `
+      :host {
+        display: inline-block;
+        max-width: min(88vw, 500px);
+      }
+      input {
+        min-height: 56px;
+      }
+    `,
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class HashtagsComponent {
+  @ViewChild(MatAutocompleteTrigger)
+  auto!: MatAutocompleteTrigger;
+  @ViewChild('hashtagInput')
+  hashtagInput!: ElementRef<HTMLInputElement>;
+  private snackBarService = inject(SnackBarService);
+  private maxChars = 50;
+  private maxHashtagsCount = 100;
+  maxAutocompleteOptions = 5;
+  separatorKeysCodes: number[] = [ENTER, COMMA];
+  hashtagCtrl = new FormControl('', Validators.maxLength(this.maxChars));
+  filteredHashtags$: Observable<string[]>;
+  addedHashtags: string[] = [];
+  allHashtags: string[] = [
+    'Impreza',
+    'Piwo',
+    'Piłkarzyki',
+    'Pizza',
+    'Pizza 2',
+    'Pizza 3',
+    'Pizza 4',
+    'Pizza 5',
+    'Karaoke',
+    'Bilard',
+    'Urodziny',
+    'BaaaaaaaaaardzoDłuuuuuugiHashtagAwogóletoooooooooo',
+    '한국말',
+  ];
+
+  constructor() {
+    this.filteredHashtags$ = this.hashtagCtrl.valueChanges.pipe(
+      startWith(null),
+      map((hashtag: string | null) =>
+        hashtag ? this._filter(hashtag) : this.allHashtags.slice()
+      )
+    );
+  }
+
+  addHashtag(event: MatChipInputEvent): void {
+    if (this.addedHashtags.length >= this.maxHashtagsCount) {
+      this.maxHashtagCountInfo();
+    } else {
+      const value = (event.value || '').trim().slice(0, this.maxChars);
```

Comment:
może  nazwać to bardziej deskryptywnie
```
      const rawHashtag = (event.value || '').trim().slice(0, this.maxChars);
```
albo
```
      const hastagStr = (event.value || '').trim().slice(0, this.maxChars);
```

Reply:
zmienione na `hashtagStr`